### PR TITLE
make call to lag-function compatible with PySpark >= 3.0.0

### PR DESCRIPTION
**Pull request by LeoAtGit**
The changed function signature was introduced with PySpark Version 3.0.0

### CODE REVIEW

File: src/pywrangler/pyspark/wranglers/interval_identifier.py [hunk 1]
```diff
@@ -539,7 +539,7 @@ def _generate_raw_iids_special(self, start_first: bool,
             ff_window)
 
         # shifting marker_col forward
-        shift_col = F.lag(forward_fill_col, default=default, count=1) \
```
Comment (mansenfranzen) [inline src/pywrangler/pyspark/wranglers/interval_identifier.py:542]: May we want to ensure backwards compatibility for `pyspark < 3.0.0` which I think is still reasonably. Hence, we will need to provide a compatibility layer around the changing keyword arguments.

**Review by mansenfranzen — CHANGES_REQUESTED** [review]
@LeoAtGit Thanks for your PR! I only have a minor request regarding backwards compatibility.

File: src/pywrangler/pyspark/wranglers/interval_identifier.py (hunk 1 above)
Comment (LeoAtGit) [reply src/pywrangler/pyspark/wranglers/interval_identifier.py:542]: For line 542 I've removed the problematic keyword entirely, as `count=1` was the default (see e.g. [here](https://spark.apache.org/docs/2.4.8/api/python/pyspark.sql.html?highlight=lag#pyspark.sql.functions.lag)) anyways, and `offset=1` is the current default (see [here](https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.functions.lag.html?highlight=lag#pyspark.sql.functions.lag))

File: src/pywrangler/pyspark/wranglers/interval_identifier.py [hunk 2]
```diff
@@ -163,7 +163,12 @@ def _drop_duplicated_marker(self, marker_column: Column, window: Window,
         mask_only = denoised == marker
 
         # use shifted column to identify subsequent duplicates
-        shifted = F.lag(denoised, count=count_lag).over(window)
+        try:
```
Comment (LeoAtGit) [inline src/pywrangler/pyspark/wranglers/interval_identifier.py:166]: There was another call to the `lag` function that used the wrong parameters, which I've now put inside of a try / except block to ensure backwards compatibility.

File: src/pywrangler/pyspark/wranglers/interval_identifier.py [hunk 3]
```diff
@@ -163,7 +163,12 @@ def _drop_duplicated_marker(self, marker_column: Column, window: Window,
...
+            # keep this code for backwards compatibility
+            shifted = F.lag(denoised, count=count_lag).over(window)
+        except TypeError:
+            # signature of the function changed in PySpark 3.0.0
+            shifted = F.lag(denoised, offset=count_lag).over(window)
```
Comment (LeoAtGit) [inline src/pywrangler/pyspark/wranglers/interval_identifier.py:171]: There was another call to the `lag` function that used the wrong parameters, which I've now put inside of a try / except block to ensure backwards compatibility.

**Pull request merged**